lib/puppet/http/service.rb

```diff
@@ -19,22 +37,48 @@ def self.create_service(client, session, name, server = nil, port = nil)
     end
   end
 
+  #
+  # Check if the service named is included in the list of available services.
+  #
+  # @param [Symbol] name
+  #
+  # @return [Boolean]
+  #
   def self.valid_name?(name)
     SERVICE_NAMES.include?(name)
   end
 
+  #
+  # Create a new service
+  #
+  # @param [Puppet::HTTP::Client] client
+  # @param [Puppet::HTTP::Session] session
+  # @param [URI] url The url to connect to
+  #
   def initialize(client, session, url)
     @client = client
     @session = session
     @url = url
   end
 
+  #
+  # Return the url with the given path encoded and appended
+  #
+  # @param [String] path the string to append to the base url
+  #
+  # @return [URI] the URI object containing the encoded path
+  #
   def with_base_url(path)
     u = @url.dup
     u.path += Puppet::Util.uri_encode(path)
     u
   end
 
+  #
+  # Open a connection using the given ssl context
+  #
+  # @param [Puppet::SSL::SSLContext] ssl_context (nil) optinal ssl context to connect with
+  #
   def connect(ssl_context: nil)
     @client.connect(@url, options: {ssl_context: ssl_context})
   end
```

Comment:
Everything below this is `protected`, so I skipped documenting it

Reply:
👍 to just public methods